### PR TITLE
fix: bump asset swapper and use equispaced samples for rust router

### PR DESCRIPTION
**Pull request by kimpers**
# Description

# Testing Instructions

# Checklist

-   [ ] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Test changes on the staging environment with [Matcha API Staging](https://api-staging.matcha.xyz).

    -   [ ] SRA/Limit orders
    -   [ ] Swap endpoints
    -   [ ] Meta transaction endpoints
    -   [ ] Depth charts

    For more information see `0x API Matcha smoke test runbook` in Quip.

### CODE REVIEW

**Comment by kimpers** [comment]
deploy bsc staging

**Comment by kimpers** [comment]
deploy polygon staging

**Comment by kimpers** [comment]
deploy polygon production

**Comment by kimpers** [comment]
deploy polygon staging

**Comment by kimpers** [comment]
deploy polygon production

**Comment by kimpers** [comment]
deploy polygon staging

**Comment by kimpers** [comment]
deploy polygon production

File: package.json [hunk 1]
```diff
@@ -97,7 +97,6 @@
         "@0x/contracts-utils": "^4.8.6",
         "@0x/dev-utils": "^4.2.10",
         "@0x/migrations": "^8.1.14",
-        "@0x/neon-router": "^0.3.0",
```
Comment (kimpers) [inline package.json:100]: This dep was unnecessary, API doesn't use it directly

**Comment by kimpers** [comment]
deploy polygon staging

**Comment by kimpers** [comment]
deploy polygon production

**Comment by kimpers** [comment]
deploy avalanche production

**Comment by kimpers** [comment]
deploy fantom production

**Comment by kimpers** [comment]
deploy celo production

**Comment by kimpers** [comment]
deploy ethereum production

**Comment by kimpers** [comment]
deploy production

**Comment by kimpers** [comment]
deploy staging

**Comment by kimpers** [comment]
deploy production

**Comment by kimpers** [comment]
deploy staging

File: yarn.lock [hunk 2]
```diff
@@ -57,9 +57,9 @@
     lodash "^4.17.11"
     valid-url "^1.0.9"
 
-"@0x/asset-swapper@^16.43.0":
-  version "16.43.0"
-  resolved "https://registry.yarnpkg.com/@0x/asset-swapper/-/asset-swapper-16.43.0.tgz#03c81eb1d70eddae40c8b7f800ba7948c57bbbd2"
+"@0x/asset-swapper@0xProject/gitpkg-registry#0x-asset-swapper-v16.44.0-37ce847d5":
```
Comment (kimpers) [inline yarn.lock:60]: **TODO** swap out for a new proper publish when testing is done

File: package.json [hunk 3]
```diff
@@ -140,7 +139,7 @@
     "dependencies": {
         "@0x/api-utils": "^0.0.1",
         "@0x/assert": "^3.0.30",
-        "@0x/asset-swapper": "^16.43.0",
+        "@0x/asset-swapper": "0xProject/gitpkg-registry#0x-asset-swapper-v16.44.0-37ce847d5",
```
Comment (kimpers) [inline package.json:142]: **TODO** Swap out for a public publish once testing is done

**Comment by kimpers** [comment]
deploy staging

**Comment by shawnxin89** [comment]
Nice!

**Review by shawnxin89 — APPROVED** [review]

**Comment by kimpers** [comment]
deploy staging

**Comment by kimpers** [comment]
deploy bsc production

**Comment by kimpers** [comment]
deploy polygon production

**Pull request merged**